framework/src/outputs/Output.C

```diff
@@ -254,7 +258,12 @@ Output::onInterval()
 
   // Update sync times if a sync time object is in use
   if (_sync_times_object)
-    _sync_times = _sync_times_object->getUniqueTimes();
+  {
+    const auto & sync_times = _sync_times_object->getUniqueTimes();
+    if (sync_times != _sync_times)
+      mooseError("The provided sync times object has changing time values. Only static time "
+                 "values are supported at this time.");
```

Comment:
```suggestion
      mooseError("The provided sync times object has changing time values. Only static time "
                 "values are supported since time steppers take sync times from the output warehouse which determines its sync times at output construction time.");
```